python/pyspark/worker.py

```diff
@@ -1890,6 +1890,14 @@ def process():
             try:
                 serializer.dump_stream(out_iter, outfile)
             finally:
+                # Sending a signal to TransformWithState UDF to perform proper cleanup steps.
```

Comment:
@HeartSaVioR @jingz-db I made a change for properly calling `close()` and other cleanup steps, could you help take a look and see if this change makes sense? I've verified with both manual test and exiting unit test to confirm this change works as expected. I'll fix the merge conflict issue later, just wanted to get some early feedbacks on this specific change, thanks!

Reply:
Looks OK to me; I assume you've confirmed that the close method is called.

Reply:
Yep, confirmed.

Reply:
LGTM! Thanks for making this change.